### DISCUSSION TITLE
Handling colons in extra metadata values

### DISCUSSION THREAD

**Discussion by samuelallan72** [todotxt/todo.txt#64] [General]
From the spec:

> Both key and value must consist of non-whitespace characters, which are not colons. Only one colon separates the key and value.

Why must a value not contain a colon?  A colon in the value doesn't appear to introduce ambiguity.  For example `label:one:two` can only be parsed as a tag with key `label` and value `one:two`.

How do most tools handle this case?  Allow it, raise an error, treat the string as part of the description since it's an invalid tag, or only parse `label:one` as the key:value, and leave `:two` in the description?

**Comment by clach04** [comment]
@swalladge "I don't know", is my correct answer from me ;-) but some thoughts.

The spec really needs spaces in between items, it explicitly says non-whitespace so I think you run a low risk of problems if you choose to use colons in your value (you deserve headaches if using in a key 😝 ) which would technically be non-conforming.

No one has shot down your idea so might be worth opening an issue as a proposal to allow. I opened https://github.com/todotxt/todo.txt/issues/77 as I personally think keys should not start with a number.

**Reply by samuelallan72** [reply]
Thanks, I opened a proposal at https://github.com/todotxt/todo.txt/pull/79
